src/titiler/core/tests/test_case_middleware.py

```diff
@@ -2,71 +2,70 @@
 
 from typing import List
 
+import pytest
 from fastapi import FastAPI, Query
 from starlette.testclient import TestClient
 from typing_extensions import Annotated
 
 from titiler.core.middleware import LowerCaseQueryStringMiddleware
 
 
-def test_lowercase_middleware():
-    """Make sure upper and lower case QS are accepted."""
+@pytest.fixture
+def app():
+    """Create test FastAPI app with LowerCaseQueryStringMiddleware."""
     app = FastAPI()
+    app.add_middleware(LowerCaseQueryStringMiddleware)
+    return app
+
+
+@pytest.fixture
+def client(app):
+    """Create test client."""
+    return TestClient(app)
+
+
+def test_lowercase_middleware(app, client):
```

Comment:
We don't need the `app` fixture here then

Reply:
ah no we need it, But then it's a bit weird that we register the endpoint after creating the application 😬

Reply:
That's correct, we can leave it as it is and close this PR if the change doesn't seem clean enough.

Reply:
Yeah I'm not convinced we need this change, sorry 🙏